tools/project.py

```diff
@@ -1388,6 +1414,253 @@ def unix_path(input: Any) -> str:
+        ("-I+", "-I"),
+        # defines
+        ("-d ", "-D"),
+        ("-D ", "-D"),
+        ("-D+", "-D"),
+    )
+
+    # Flags to pass through
+    CFLAG_PASSTHROUGH: Set[str] = set()
+    CFLAG_PASSTHROUGH_PREFIX: Tuple[str, ...] = (
+        "-I",  # includes
+        "-D",  # defines
+    )
+
+    # Defines for specific target platforms
+    GC_DEFINES: Tuple[str, ...] = ("__PPCGEKKO__",)
+    WII_DEFINES: Tuple[str, ...] = (
+        *GC_DEFINES,
+        "__PPCBROADWAY__",
+    )
+
+    def gc_defines(mwerks: str) -> Tuple[str, ...]:
+        return (*GC_DEFINES, f"__MWERKS__={mwerks}")
+
+    def gc_defines_2(mwerks: str, cwbuild: str) -> Tuple[str, ...]:
+        return (
+            *GC_DEFINES,
+            f"__MWERKS__={mwerks}",
+            f"__CWCC__={mwerks}",
+            f"__CWBUILD__={cwbuild}",
+        )
+
+    def wii_defines(mwerks: str, cwbuild: str) -> Tuple[str, ...]:
+        return (
+            *WII_DEFINES,
+            f"__MWERKS__={mwerks}",
+            f"__CWCC__={mwerks}",
+            f"__CWBUILD__={cwbuild}",
+        )
+
+    # Defines for specific compiler versions
+    # Based on a lot of manual testing
+    COMPILER_DEFINES: Dict[str, Tuple[str, ...]] = {
+        "GC/1.0": gc_defines("0x2301"),
+        "GC/1.1": gc_defines("0x2301"),
+        "GC/1.2.5": gc_defines("0x2301"),
+        "GC/1.2.5e": gc_defines("0x2301"),
+        "GC/1.2.5n": gc_defines("0x2301"),
+        "GC/1.3": gc_defines("0x2406"),
+        "GC/1.3.2": gc_defines("0x2407"),
+        "GC/1.3.2r": gc_defines("0x2407"),
+        "GC/2.0": gc_defines("0x2407"),
+        "GC/2.0p1": gc_defines("0x2407"),
+        "GC/2.5": gc_defines("0x2407"),
+        "GC/2.6": gc_defines("0x2407"),
+        "GC/2.7": gc_defines("0x2407"),
+        "GC/3.0a3": gc_defines_2("0x4100", "51213"),
+        "GC/3.0a3.2": gc_defines_2("0x4200", "60126"),
+        "GC/3.0a3.3": gc_defines_2("0x4200", "60289"),
+        "GC/3.0a3.4": gc_defines_2("0x4200", "60308"),
+        "GC/3.0a5": gc_defines_2("0x4200", "60422"),
+        "GC/3.0a5.2": wii_defines("0x4199", "60831"),
+        "GC/3.0": wii_defines("0x4199", "60831"),
+        "Wii/0x4201_127": wii_defines("0x4201", "142"),
+        "Wii/1.0RC1": wii_defines("0x4201", "140"),
+        "Wii/1.0a": wii_defines("0x4201", "142"),
+        "Wii/1.0": wii_defines("0x4302", "145"),
+        "Wii/1.1": wii_defines("0x4302", "151"),
+        "Wii/1.3": wii_defines("0x4302", "172"),
+        "Wii/1.5": wii_defines("0x4302", "188"),
+        "Wii/1.6": wii_defines("0x4302", "202"),
+        "Wii/1.7": wii_defines("0x4302", "213"),
+    }
+
+    clangd_config = []
+
+    def add_unit(build_obj: Dict[str, Any]) -> None:
+        obj = objects.get(build_obj["name"])
+        if obj is None:
+            return
+
+        # Skip unresolved objects
+        if (
+            obj.src_path is None
+            or obj.src_obj_path is None
+            or obj.src_path.suffix not in (".c", ".cp", ".cpp")
```

Comment:
```suggestion
            or obj.src_path.suffix == ".o"
```
Maybe? Because there's also `.C` and some others I'm probably forgetting.

Reply:
I copied that from `generate_build_ninja`; just `== ".o"` wouldn't be sufficient, as that wouldn't exclude `.s` files. Probably better to make a global for allowed C/C++ extensions and apply that to both places.

Reply:
Agreed, that logic should be centralized at some point

Reply:
Added some helper functions for checking file extensions to address this.